docs/auth.md

```diff
@@ -34,142 +33,43 @@ In a full setup of the service catalog API server, there are three
    the service catalog serving certificates, but a different CA
    may also be used.
 
-   By default, the service catalog API server automatically generates
-   self-signed certificates if no serving certificates are passed in,
-   making this CA optional.  However, in a real setup, you'll need this CA
-   so that clients can easily trust the identity of the service catalog
-   API server.
+   The service catalog Helm chart automatically generates this CA. There is
+   generally no need to override this.
```

Comment:
I would qualify this statement like "no need to override this unless you have special requirements like using an existing CA"

Reply:
I can add that, however, as this PR stands, the option to provide your own isn't present. Truthfully, I tried to think of a scenario where it was inadequate to use the one that is generated and I couldn't come up with one.

Reply:
Since I made the comment above from my phone, I didn't fully qualify it. I will do so now...

The chart generates a CA and uses it to sign just one cert. These are used to establish trust between just two parties-- the aggregator and the svc cat apiserver. It's not used for any other purpose. With this narrow use understood, I've asked myself what benefit there could possibly be to providing your own CA. The only thing I can come up with is the possibility that you wish for a client other than the aggregator to trust the svc cat apiserver...

I see that as an extreme outlying use case. Nothing stops the svc catalog software from using whatever cert you give it, signed by whatever CA, and likewise, nothing stops the aggregator from using whatever CA you give it. However, I can't see any reason that the chart needs to have this degree of flexibility to accommodate such a fringe case.

It's possible I am wrong, but I'd rather see us do the simple thing first and circle back to add complexity only if and when it becomes clear that there's a good reason for it.

Reply:
+1 for simplifying the existing process and adding complexity if the need for it arises.

Reply:
SGTM, we can add it back if we need to later.